server/modules/taxonomy/repository/taxonomyRepository.js

```diff
@@ -34,19 +34,22 @@ export const insertTaxonomy = async (surveyId, taxonomy, client = db) =>
   )
 
 const _insertOrUpdateVernacularNames = (surveyId, taxonUuid, vernacularNames, client = db) =>
-  Object.entries(vernacularNames).map(([lang, vernacularName]) =>
-    client.none(
-      `INSERT INTO 
+  R.pipe(R.values, R.flatten, R.map(vernacularName =>
+      client.none(
+        `INSERT INTO 
            ${getSurveyDBSchema(surveyId)}.taxon_vernacular_name (uuid, taxon_uuid, props_draft)
         VALUES 
           ($1, $2, $3)
-        ON CONFLICT 
-          (taxon_uuid, ((props||props_draft)->>'${TaxonVernacularName.keysProps.lang}')) DO
-        UPDATE SET 
+        ON CONFLICT (
+          taxon_uuid, 
+          ((props||props_draft)->>'${TaxonVernacularName.keysProps.lang}'), 
+          ((props||props_draft)->>'${TaxonVernacularName.keysProps.name}')
+        ) DO UPDATE SET 
           props_draft = ${getSurveyDBSchema(surveyId)}.taxon_vernacular_name.props_draft || $3`,
-      [TaxonVernacularName.getUuid(vernacularName), taxonUuid, TaxonVernacularName.getProps(vernacularName)]
+        [TaxonVernacularName.getUuid(vernacularName), taxonUuid, TaxonVernacularName.getProps(vernacularName)]
+      )
     )
-  )
```

Comment:
are you updating the name of a vernacular name?
e.g first import i have 'ita':'name1'
second import i have 'ita':'name111' and 'ita':'name112'.
what do you get in the end?

Reply:
name1, name111, name112. 
You never delete or update existing vernacular names (for published taxonomies).
For non published taxonomies we delete everything before importing a new list.

Reply:
you need to be able to update an existing vernacular name.
e.g first import i have 'ita':'name1'
second import i have 'ita':'name111' and 'ita':'name112'.
in the end you need to have the result of the second import

Reply:
I'm not sure about this. If a user used that vernacular name, we should not change it into something different... or do we?!

Reply:
if you can change a scientific name you can change a vernacular name. why should vernacular name work differently?
Don't you store the uuid in the node?

Reply:
uhm, right, I'll change it